groups/bdl/bdlb/bdlb_guid.t.cpp

```diff
@@ -407,6 +409,46 @@ int main(int argc, char *argv[])
         }
 //..
       } break;
+      case 15: {
+        // --------------------------------------------------------------------
+        // TESTING HASH FUNCTION
+        //
+        // Concerns:
+        //: 1 A guid object can be hashed by instances of the 'bslh::Hash'
+        //:   template and 'bsl::hash<bdlb::Guid>'.
+        //:
+        //: 2 A small sample of different guid objects produce different
+        //:   hashes.
+        //
+        // Plan:
+        //: 1 Hash nine different guid objects and verify 'bslh::Hash<>'
+        //:   produces the same result as 'bsl::hash<bdlb::Guid>'
+        //:
+        //: 2 Hash nine different guid objects and verify they produce
+        //:   nine different hashes.
+        //
+        // Testing:
+        //   bsl::hash<bdlb::Guid>
+        //   bslh::Hash<>
+        // --------------------------------------------------------------------
+        if (verbose) cout << endl
+                          << "TESTING HASH FUNCTION" << endl
+                          << "=====================" << endl;
+
+        const bsl::size_t NUM_VALUES = sizeof(VALUES) / sizeof(VALUES[0]);
+
+        bsl::hash<Obj> bsl_hash_function;
+        bslh::Hash<>   bslh_hash_function;
+
+        bsl::set<bsl::size_t> hash_results;
+
+        for (bsl::size_t i = 0; i < NUM_VALUES; ++i) {
+            Obj guid(VALUES[i]);
+            ASSERT(bsl_hash_function(guid) == bslh_hash_function(guid));
```

Comment:
Maybe it'd be more useful to test `bsl::hash<Obj>` against the hash algorithm directly, rather than `bslh::Hash` (as a verification that `hashAppend` is calling the algorithm correctly).  I think this is just verifying `bsl::hash<Obj>` is calling `bsl::Hash<>`, which isn't related to `Guid`. (Admittedly haven't done any results testing beyond uniqueness for other hash functions we've implemented).